Product/Production/Services/AuditRepositoryCore/src/main/java/gov/hhs/fha/nhinc/audit/transform/AuditTransform.java

```diff
@@ -0,0 +1,608 @@
+            LOG.error("DatatypeConfigurationException when creating XMLGregorian Date");
+            LOG.error(" message: " + e.getMessage(), e);
+        } catch (ArrayIndexOutOfBoundsException e) {
+            LOG.error("ArrayIndexOutOfBoundsException when creating XMLGregorian Date");
+            LOG.error(" message: " + e.getMessage(), e);
+        }
+        // Set the Event Outcome Indicator
+        BigInteger eventOutcomeBig = BigInteger.ZERO;
+        if (eventOutcome != null) {
+            eventOutcomeBig = new BigInteger(eventOutcome.toString());
+        }
+        eventIdentification.setEventOutcomeIndicator(eventOutcomeBig);
+
+        // Set the Event Id
+        eventIdentification.setEventID(eventId);
+
+        return eventIdentification;
+    }
+
+    /**
+     * Create the <code>CodedValueType</code> for an audit log record.
+     *
+     * @param code
+     * @param codeSys
+     * @param codeSysName
+     * @param dispName
+     * @return <code>CodedValueType</code>
+     */
+    private static CodedValueType createCodeValueType(String code, String codeSys, String codeSysName, String dispName) {
+        CodedValueType codeValueType = new CodedValueType();
+
+        // Set the Code
+        if (NullChecker.isNotNullish(code)) {
+            codeValueType.setCode(code);
+        }
+
+        // Set the Codesystem
+        if (NullChecker.isNotNullish(codeSys)) {
+            codeValueType.setCodeSystem(codeSys);
+        }
+
+        // Set the Codesystem Name
+        if (NullChecker.isNotNullish(codeSysName)) {
+            codeValueType.setCodeSystemName(codeSysName);
+        }
+
+        // Set the Display Name
+        if (NullChecker.isNotNullish(dispName)) {
+            codeValueType.setDisplayName(dispName);
+        }
+
+        return codeValueType;
+    }
+
+    /**
+     * Create an <code>AuditSourceIdentificationType</code> based on the community id and name for an audit log record.
+     *
+     * @param communityId
+     * @param communityName
+     * @return <code>AuditSourceIdentificationType</code>
+     */
+    private static AuditSourceIdentificationType createAuditSourceIdentification(String communityId, String communityName) {
+        AuditSourceIdentificationType auditSrcId = new AuditSourceIdentificationType();
+
+        // Set the Audit Source Id (community id)
+        if (communityId != null) {
+            if (communityId.startsWith("urn:oid:")) {
+                auditSrcId.setAuditSourceID(communityId.substring(8));
+            } else {
+                auditSrcId.setAuditSourceID(communityId);
+            }
+        }
+
+        // If specified, set the Audit Enterprise Site Id (community name)
+        if (communityName != null) {
+            auditSrcId.setAuditEnterpriseSiteID(communityName);
+        }
+
+        return auditSrcId;
+    }
+    
+    private LogEventRequestType createAuditMessage(Object msg, AssertionType assertion,
```

Comment:
This needs to be refactored.  Need to either use generics or objects.  Not both.